pkg/api/endpoints/util.go

```diff
@@ -64,12 +65,37 @@ func RepackSubsets(subsets []api.EndpointSubset) []api.EndpointSubset {
 				}
 				portsToAddrs[*epp].Insert(allAddrs[ak])
 			}
+			for k := range subsets[i].NotReadyAddresses {
```

Comment:
it feels like there is a lot of code duplication here with what is in subsets[i].Addresses, would be good if we could somehow make something common.